### PR TITLE
[Tune] Transformer blog example

### PR DESCRIPTION
**Pull request by amogkam**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: ci/jenkins_tests/run_tune_tests.sh [hunk 1]
```diff
@@ -162,6 +162,10 @@ $SUPPRESS_OUTPUT docker run --rm --shm-size=${SHM_SIZE} --memory=${MEMORY_SIZE}
     python /ray/python/ray/tune/examples/pbt_dcgan_mnist/pbt_dcgan_mnist.py \
     --smoke-test
 
+$SUPPRESS_OUTPUT docker run --rm --shm-size${SHM_SIZE} --memory=${MEMORY_SIZE} --memory-swap=-1 $DOCKER_SHA \
+    python /ray/python/ray/tune/examples/pbt_transformers/pbt_transformers.py \
+    --smoke-test
+
```
Comment (richardliaw) [inline ci/jenkins_tests/run_tune_tests.sh:168]: btw this file is deprecated; you should use `tune/BUILD` instead. See https://github.com/ray-project/ray/pull/9609/files for the change

File: python/ray/tune/examples/README.rst [hunk 2]
```diff
@@ -52,6 +52,11 @@ LightGBM Example
 
 - `lightgbm_example <https://github.com/ray-project/ray/blob/master/python/ray/tune/examples/lightgbm_example.py>`__: Trains a basic LightGBM model with Tune with the function-based API and a LightGBM callback.
 
+:hugging_face: Transformers Example
```
Comment (richardliaw) [inline python/ray/tune/examples/README.rst:55]: does this actually work? interesting

File: python/ray/tune/examples/pbt_transformers/pbt_transformers.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
+# flake8: noqa
+# yapf: disable
+
+import os
+import ray
+from ray.tune import CLIReporter
+from ray.tune.schedulers import PopulationBasedTraining
+
+from ray import tune
+import trainer
```
Comment (richardliaw) [inline python/ray/tune/examples/pbt_transformers/pbt_transformers.py:10]: Can you put everything into one file?

Comment (richardliaw) [reply python/ray/tune/examples/pbt_transformers/pbt_transformers.py:10]: actually this is fine - but please be sure to have absolute imports

File: ci/jenkins_tests/run_tune_tests.sh (hunk 1 above)
Comment (amogkam) [reply ci/jenkins_tests/run_tune_tests.sh]: added it to BUILD as well.

File: python/ray/tune/examples/pbt_transformers/pbt_transformers.py (hunk 3 above)
Comment (amogkam) [reply python/ray/tune/examples/pbt_transformers/pbt_transformers.py]: done

File: python/ray/tune/examples/README.rst (hunk 2 above)
Comment (amogkam) [reply python/ray/tune/examples/README.rst]: I got it working for the sphinx docs but cant figure out how to do it for Github rst

**Review by richardliaw — APPROVED** [review]

File: doc/source/tune/examples/pbt_transformers.rst [hunk 4]
```diff
@@ -0,0 +1,6 @@
+:orphan:
+
+pbt_transformers_example
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. literalinclude:: /../../python/ray/tune/examples/pbt_transformers/pbt_transformers.py
```
Comment (richardliaw) [inline doc/source/tune/examples/pbt_transformers.rst:6]: Can you also literalinclude all of pbt_transformers/ files?

**Review by richardliaw — COMMENTED** [review]
Code looks awesome. Can you add a README.rst in `pbt_transformers/` so that users can know how to run things and such?

File: python/ray/tune/BUILD [hunk 5]
```diff
@@ -474,6 +474,16 @@ py_test(
 #     args = ["--smoke-test"]
 # )
 
+py_test(
+    name = "pbt_transformers_example",
+    size = "medium",
+    srcs = ["examples/pbt_transformers/pbt_transformers.py"],
```
Comment (richardliaw) [inline python/ray/tune/BUILD:480]: I think you need to add `transformers` into the right requirements txt file? 

And perhaps also the documentation txt file?

File: doc/source/tune/examples/pbt_transformers.rst (hunk 4 above)
Comment (amogkam) [reply doc/source/tune/examples/pbt_transformers.rst]: done.

File: python/ray/tune/BUILD (hunk 5 above)
Comment (amogkam) [reply python/ray/tune/BUILD:480]: Added transformers to tune dependencies.

Comment (amogkam) [reply python/ray/tune/BUILD:480]: Why would I need to add transformers to the requirements-doc.txt? Or are you referring to something else?

File: python/ray/tune/examples/pbt_transformers/pbt_transformers.py [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+                     gpus_per_trial=0,
+                     smoke_test=False,
+                     ray_address=None):
+    ray.init(ray_address, log_to_driver=False)
+    data_dir = os.path.abspath(os.path.join(os.getcwd(), "./data"))
+    if not os.path.exists(data_dir):
+        os.mkdir(data_dir, 0o755)
+
+    # Change these as needed.
+    model_name = "bert-base-uncased"
```
Comment (richardliaw) [inline python/ray/tune/examples/pbt_transformers/pbt_transformers.py:132]: for smoke-test can we actually use a super tiny model?

**Pull request merged**